File: 2ndLargestElement.py
n = int(input("Enter number of elements: "))

arr = []
print("Enter array elements:")
for _ in range(n):
    arr.append(int(input()))

largest = float("-inf")
second_largest = float("-inf")

for num in arr:
    if num > largest:
        second_largest = largest
        largest = num
    elif num > second_largest and num != largest:
        second_largest = num

if second_largest == float("-inf"):
    print("No second largest element found.")
else:
    print("Second largest element:", second_largest)

File: 2ndLargestElement.cpp
#include <iostream>
#include <climits>
using namespace std;

int main() {
    int n;

    cout << "Enter number of elements: ";
    cin >> n;

    int arr[n];
    cout << "Enter array elements:\n";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int largest = INT_MIN;
    int secondLargest = INT_MIN;

    for (int i = 0; i < n; i++) {
        int num = arr[i];

        if (num > largest) {
            secondLargest = largest;
            largest = num;
        } else if (num > secondLargest && num != largest) {
            secondLargest = num;
        }
    }

    if (secondLargest == INT_MIN) {
        cout << "No second largest element found.";
    } else {
        cout << "Second largest element: " << secondLargest;
    }

    return 0;
}
